libs/remix-core-plugin/src/lib/compiler-artefacts.ts

```diff
@@ -72,51 +72,61 @@ export class CompilerArtefacts extends Plugin {
-          const artifacts = JSON.parse(content)
-          return { abi: artifacts.abi, bytecode: artifacts.data.bytecode.object }
-        } else {
-          for (const dirPath of dirList) {
-            const result = await this.getArtefactsFromFE (dirPath, contractName)
-            if (result) return result
+      for (const dirPath of dirList) {
+        if(dirPath === path + '/artifacts' && await this.call('fileManager', 'exists', dirPath + '/build-info')) {
+          const buildFileList = await this.call('fileManager', 'fileList', dirPath + '/build-info')
+          for (const buildFile of buildFileList) {
+            let content = await this.call('fileManager', 'readFile', buildFile)
+            if (content) content = JSON.parse(content)
+            const contracts = content.output.contracts
+            const artefacts = this._getAllContractArtefactsfromOutput(contracts, contractName)
+            Object.assign(contractArtefacts, artefacts)
           }
-        }
-      } else {
-        for (const dirPath of dirList) {
-          const result = await this.getArtefactsFromFE (dirPath, contractName)
-          if (result) return result
-        }
-      }
+        } else await this._getArtefactsFromFE (dirPath, contractName, contractArtefacts)
+      } 
     } else return
   }
 
   async getArtefactsByContractName (contractName) {
     const contractsDataByFilename = this.getAllContractDatas()
-    const contractsData = Object.values(contractsDataByFilename)
-    if (contractsData && contractsData.length) {
-      const index = contractsData.findIndex((contractsObj) => Object.keys(contractsObj).includes(contractName))
-      if (index !== -1) return { abi: contractsData[index][contractName].abi, bytecode: contractsData[index][contractName].evm.bytecode.object }
+    if (contractName.includes(':')) {
+      const nameArr = contractName.split(':')
+      const filename = nameArr[0]
+      const contract = nameArr[1]
+      if(Object.keys(contractsDataByFilename).includes(filename) && contractsDataByFilename[filename][contract]) 
+        return contractsDataByFilename[filename][contract]
       else {
-        const result = await this.getArtefactsFromFE ('contracts', contractName)
-        if (result) return result
+        const allContractsData = {}
+        await this._getArtefactsFromFE ('contracts', contract, allContractsData)
+        if(allContractsData[contractName]) return allContractsData[contractName]
         else throw new Error(`Could not find artifacts for ${contractName}. Compile contract to generate artifacts.`)
       }
     } else {
-      const result = await this.getArtefactsFromFE ('contracts', contractName)
-      if (result) return result
-      else throw new Error(`Could not find artifacts for ${contractName}. Compile contract to generate artifacts.`)
-    }  
+      const contractArtefacts = this._getAllContractArtefactsfromOutput(contractsDataByFilename, contractName)
```

Comment:
`_getAllContractArtefactsfromOutput` is used to get all the duplicates for a given contract? in that case it might be better to name it `_getDuplicateArtefacts` or something like that.

Reply:
Its not necessary that there will be duplicates